trio/_core/tests/test_run.py

```diff
@@ -1590,6 +1591,15 @@ def generator_based_coro():  # pragma: no cover
                 bad_call(len, [1, 2, 3])
             assert "appears to be synchronous" in str(excinfo.value)
 
+            @async_generator
+            async def async_gen(arg):
+                yield_(await sleep(arg))
```

Comment:
This line is never executed, you probably want to use the nocover + pass tricked as above.

Reply:
Gotcha. Pushed it. Hopefully that works.